dispatch/modules/content/models.py

```diff
@@ -553,3 +553,42 @@ class Issue(Model):
     volume = PositiveIntegerField(null=True)
     issue = PositiveIntegerField(null=True)
     date = DateTimeField()
+
+class Poll(Model):
+    name = CharField(max_length=255)
+    question = CharField(max_length=255)
+    is_open = BooleanField(default=True)
+    show_results = BooleanField(default=True)
+
+    def save_answers(self, answers, is_new):
+        if is_new is False:
+            self.delete_old_answers(answers)
+        for answer in answers:
+            try:
+                answer_obj = PollAnswer.objects.get(poll=self, id=answer['id'])
+                answer_obj.name = answer['name']
+            except PollAnswer.DoesNotExist:
+                answer_obj = PollAnswer(poll=self, name=answer['name'])
+            answer_obj.save()
+
+    def delete_old_answers(self, answers):
+        old_answers = PollAnswer.objects.filter(poll=self)
+        for answer in answers:
+            old_answers = old_answers.exclude(id=answer['id'])
+        old_answers.delete()
```

Comment:
if you want, you could convert the loop into an implicit `id__in`, which is likely what django is already doing in the qs internally at the time it's committed, but is maybe slightly cleaner:
```py
PollAnswer.objects.filter(poll=self) \
    .exclude(id__in=[answer['id'] for answer in answers]) \
    .delete()
```